src/mad_spark_alt/core/intent_detector.py

```
@@ -0,0 +1,219 @@
+    def __init__(self) -> None:
+        """Initialize the detector with intent patterns."""
+        
+        # Intent patterns with keywords and phrases
+        self.intent_patterns: Dict[QuestionIntent, Dict[str, List[str]]] = {
+            QuestionIntent.ENVIRONMENTAL: {
+                "keywords": [
+                    "environment", "environmental", "sustainable", "sustainability",
+                    "plastic", "waste", "pollution", "climate", "carbon", "emissions",
+                    "recycle", "recycling", "green", "eco", "ecological", "nature",
+                    "conservation", "biodegradable", "renewable", "energy", "solar",
+                    "wind", "ocean", "forest", "ecosystem", "habitat", "species"
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)